ui/src/alerting/components/builder/CheckMatchingRulesCard.tsx

```diff
@@ -1,19 +1,86 @@
 // Libraries
-import React, {FunctionComponent} from 'react'
+import React, {FunctionComponent, useState, useEffect} from 'react'
+import {connect} from 'react-redux'
+import {get} from 'lodash'
 
 // Components
+import MatchingRuleCard from 'src/alerting/components/builder/MatchingRuleCard'
+
+// Actions
+import {getActiveTimeMachine} from 'src/timeMachine/selectors'
+
+// API
+import * as api from 'src/client'
+
+//Types
+import {NotificationRule, Check, AppState, CheckTagSet} from 'src/types'
 import {EmptyState, ComponentSize} from '@influxdata/clockface'
 
-const CheckMatchingRulesCard: FunctionComponent = () => {
+interface StateProps {
+  check: Partial<Check>
+  orgID: string
+}
+
+const CheckMatchingRulesCard: FunctionComponent<StateProps> = ({
+  check,
+  orgID,
+}) => {
+  const getMatchingRules = async (): Promise<NotificationRule[]> => {
+    const tags: CheckTagSet[] = get(check, 'tags', [])
+    const tagString = tags.map(t => `tag=${t.key}:${t.value}`).join('&')
+
+    // todo also: get tags from query results
+    const resp = await api.getNotificationRules({
+      query: {orgID, tag: tagString},
+    })
+
+    if (resp.status !== 200) {
+      // throw new Error(resp.data.message)
```

Comment:
It would be great to set an error message in local state and display it somewhere inside this component... that's really my hope for how this typed error response stuff shakes out—that we can do more than catch errors and display generic notifications.